solr/core/src/test/org/apache/solr/cloud/HttpPartitionTest.java

```diff
@@ -548,9 +548,6 @@ protected int sendDoc(int docId, Integer minRf, SolrClient solrClient, String co
     doc.addField("a_t", "hello" + docId);
 
     UpdateRequest up = new UpdateRequest();
-    if (minRf != null) {
-      up.setParam(UpdateRequest.MIN_REPFACT, String.valueOf(minRf));
-    }
```

Comment:
Hi @trdillon - thanks for opening this pull request!

Okay, I think I see it now, so your https://issues.apache.org/jira/browse/SOLR-14034?focusedCommentId=17223427&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17223427 question is about this sort of test change here i.e. whether or not `minRf` should remain an unused `sendDoc` argument or whether or not it should be removed. I'm thinking here in `HttpPartitionTest` removal makes sense (haven't yet looked at `ReplicationFactorTest`). What do you think?

Reply:
@cpoerschke Thanks for all your help with this one.

`sendDoc` seems OK to remove from `HttpPartitionTest` as it's only passed `minRf` as an argument, but in `ReplicationFactorTest` it is passed `expectedRf` as an argument in `addDocs`:
https://github.com/apache/lucene-solr/blob/98f12f4aeb9f6ec9f5c4de53f9faddc41043df59/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java#L417

`sendDocsWithRetry`  is implemented here and used in a few different test classes (`DistributedVersionInfoTest`, `LeaderFailoverAfterPartitionTest` and `ForceLeaderTest`) with `minRf` as an argument:
https://github.com/apache/lucene-solr/blob/98f12f4aeb9f6ec9f5c4de53f9faddc41043df59/solr/test-framework/src/java/org/apache/solr/cloud/AbstractFullDistribZkTestBase.java#L941

But it's also implemented using `expectedRfDBQ` and `expectedRf` in `ReplicationFactorTest`:

- https://github.com/apache/lucene-solr/blob/98f12f4aeb9f6ec9f5c4de53f9faddc41043df59/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java#L230
- https://github.com/apache/lucene-solr/blob/98f12f4aeb9f6ec9f5c4de53f9faddc41043df59/solr/core/src/test/org/apache/solr/cloud/ReplicationFactorTest.java#L427

I wasn't sure how to proceed with this as looking at the tests it seems that `expectedRf` is relied upon quite often.

Reply:
Thanks for the code pointers and detailed explanation!

How about we scope this pull request to remove `minRf` in almost all code paths i.e. `sendDoc` gets its unused argument removed in both the replication factor and http partition tests but `sendDocsWithRetry` remains unchanged for now since as you say it's used in other test classes too?

It appears that the `minRf` argument in `sendDocsWithRetry` is indeed unused already but it might be clearer to tidy that up in a separate follow-up pull request rather than mix it in here at this time.

Reply:
> It appears that the minRf argument in sendDocsWithRetry is indeed unused already but it might be clearer to tidy that up in a separate follow-up pull request rather than mix it in here at this time.

I'm wondering if it would be better for this method to actually start using this parameter, and retry in the case of `minRf` not achieved? This is not something that was being done in this method, but may help improve some of the tests using it. I don't think such a change should be done in this PR, but maybe lets not remove the parameter yet then. WDYT?

Reply:
> I'm wondering if it would be better for this method to actually start using this parameter, ...

Ah, yes, I agree, if it's a case of a not-yet-used parameter then keeping it and starting to use it makes sense. Removal would only be for no-longer-used parameter cases. Good catch!